File: input.c
#include <wiringPi.h>

int main()
{
 wiringPiSetupGpio(); //Broadcom pin system
 pinMode(23, OUTPUT); //initialize pin as output
 pinMode(24, INPUT); //initialize pin as output


 while(1)
{
	if (digitalRead(24)) // Button not presses gives 1
        {
                   digitalWrite(23, LOW);     // Regular LED off
        }
	else
   	{
		digitalWrite(23, HIGH);
		delay(25);
 	}


}

}

File: input.py
import RPi.GPIO as GPIO
import time

GPIO.setmode(GPIO.BCM)
GPIO.setup(23, GPIO.OUT)
GPIO.setup(24, GPIO.IN)

print("helo press the button to on the LED")

try:
 while 1:
  if GPIO.input(24):
   GPIO.output(23, GPIO.LOW)
  else:
   GPIO.output(23, GPIO.HIGH)

except KeyboardInterrupt:
 GPIO.cleanup()
